pkg/image/mutable_source.go

```diff
@@ -177,28 +177,29 @@ func (m *MutableSource) Env() map[string]string {
 
 // SetEnv takes a map of environment variables, and converts them to an array of strings
 // in the form KEY=VALUE, and then sets the image config
-func (m *MutableSource) SetEnv(envMap map[string]string, author string) {
+func (m *MutableSource) SetEnv(envMap map[string]string, author, createdBy string) {
 	envArray := []string{}
 	for key, value := range envMap {
 		entry := key + "=" + value
 		envArray = append(envArray, entry)
 	}
 	m.cfg.Schema2V1Image.Config.Env = envArray
-	m.AppendConfigHistory(author, true)
+	m.AppendConfigHistory(author, createdBy, true)
 }
 
 func (m *MutableSource) Config() *manifest.Schema2Config {
 	return m.cfg.Schema2V1Image.Config
 }
 
-func (m *MutableSource) SetConfig(config *manifest.Schema2Config, author string, emptyLayer bool) {
+func (m *MutableSource) SetConfig(config *manifest.Schema2Config, author, createdBy string, emptyLayer bool) {
```

Comment:
Maybe refactor this a bit to take a manifest.Schema2History struct instead of all the fields as parameters.